### PR TITLE
Rename model to system

### PR DESCRIPTION
**Pull request by richfitz**
~Merge after #26, contains those commits~

This PR is a mass rename from "model" to "system" as I felt the former was painfully overloaded once we started talking about mcstate models (which are statistical models).  System feels largely ok as we have a "system of equations".

### CODE REVIEW

File: R/compile.R [hunk 1]
```diff
@@ -113,7 +113,7 @@ dust_compile <- function(filename, quiet = FALSE, workdir = NULL,
 
 dust_generate <- function(config, filename, linking_to, cpp_std,
                           optimisation_level, compiler_options, mangle) {
-  model <- read_lines(filename)
+  system <- read_lines(filename)
```
Comment (weshinsley) [inline R/compile.R:116]: Hmmm - lines like this will take a little bit of getting used to!

File: R/zzz.R [hunk 2]
```diff
@@ -1,3 +1,3 @@
 ##' @useDynLib dust2, .registration = TRUE
-cache <- list2env(list(models = list()),
+cache <- list2env(list(generators = list()),
```
Comment (weshinsley) [inline R/zzz.R:2]: A cache used to have models, but now has generators, rather than systems...

**Review by weshinsley — APPROVED** [review]
All good - will take a little getting used to on the aesthetic front - I'm pre-disposed to thinking of "system" in a particular way from other languages, so occasionally it looks a little out of place until I adjust!

(but I don't think we want to be specifically writing "dust system" every time we refer to it either as that would get quite tiring to write/read... )

File: R/zzz.R (hunk 2 above)
Comment (richfitz) [reply R/zzz.R:2]: this really should have been generators previously too

**Pull request merged**